### PR TITLE
Add tests for the cookies example

### PR DESCRIPTION
**Pull request by sethlopezme**
Relates to #62.

### CODE REVIEW

**Comment by sethlopezme** [comment]
The issue you brought up is now fixed

File: examples/cookies/src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    assert_eq!(response.body().and_then(|b| b.into_string()), Some(expected_body));
+}
+
+#[test]
+fn test_index() {
+    // Render the template with an empty context to test against.
+    let mut context: HashMap<&str, &str> = HashMap::new();
+    let template = Template::render("index", &context);
+    // Test the route without sending the "message" cookie.
+    test_body(None, format!("{}", template));
```
Comment (SergioBenitez) [inline examples/cookies/src/tests.rs]: Use `template.to_string()` instead of `format!`.

File: examples/cookies/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    // Render the template with a context that contains the message.
+    context.insert("message", "Hello from Rocket!");
+    let template = Template::render("index", &context);
+    // Test the route with the "message" cookie.
+    test_body(Some(Cookie::new("message".to_string(),
+                               "Hello from Rocket!".to_string())),
+              format!("{}", template));
```
Comment (SergioBenitez) [inline examples/cookies/src/tests.rs]: Same here.

File: examples/cookies/src/tests.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    let rocket = rocket::ignite().mount("/", routes![super::index]);
+    let mut request = MockRequest::new(Method::Get, "/");
+
+    // Attach a cookie if one is given.
+    if let Some(cookie) = optional_cookie {
+        request = request.cookie(cookie);
+    }
+
+    let mut response = request.dispatch_with(&rocket);
+
```
Comment (SergioBenitez) [inline examples/cookies/src/tests.rs]: Don't need this empty line here.

File: examples/cookies/src/tests.rs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    assert_eq!(response.status(), Status::Ok);
+    assert_eq!(response.body().and_then(|b| b.into_string()), Some(expected_body));
+}
+
+#[test]
+fn test_index() {
+    // Render the template with an empty context to test against.
+    let mut context: HashMap<&str, &str> = HashMap::new();
+    let template = Template::render("index", &context);
+    // Test the route without sending the "message" cookie.
```
Comment (SergioBenitez) [inline examples/cookies/src/tests.rs]: Add an empty lake above this comment. I find empty lines above comments to be much more readable.

Comment (SergioBenitez) [inline examples/cookies/src/tests.rs]: Line*

File: examples/cookies/src/tests.rs [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    test_body(None, format!("{}", template));
+
+    // Render the template with a context that contains the message.
+    context.insert("message", "Hello from Rocket!");
+    let template = Template::render("index", &context);
+    // Test the route with the "message" cookie.
```
Comment (SergioBenitez) [inline examples/cookies/src/tests.rs]: Same above here.

**Comment by sethlopezme** [comment]
Your requested changes have been implemented!

**Comment by SergioBenitez** [comment]
Merged in da7026c78140db823c39d4b1391cb3df5eeb9c0f.

**Pull request closed without merging**